indico/web/client/js/jquery/widgets/jinja/principal_list_widget.js

```diff
@@ -1,28 +1,49 @@
-// This file is part of Indico.
-// Copyright (C) 2002 - 2020 CERN
-//
-// Indico is free software; you can redistribute it and/or
-// modify it under the terms of the MIT License; see the
-// LICENSE file for more details.
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {WTFPrincipalListField} from 'indico/react/components/WTFPrincipalListField';
 
 (function(global) {
   'use strict';
 
   global.setupPrincipalListWidget = function setupPrincipalListWidget(options) {
-    options = $.extend(
-      true,
-      {
+    options = {
+      ...{
         fieldId: null,
         eventId: null,
         openImmediately: false,
-        groups: null,
-        allowExternal: false,
-        allowNetworks: false,
+        withGroups: false,
+        withExternalUsers: false,
+        withEventRoles: false,
+        withCategoryRoles: false,
+        withRegistrationForms: false,
+        withNetworks: false,
         networks: [],
       },
-      options
+      ...options,
+    };
+    return options.legacy
+      ? _setupLegacyPrincipalListWidget(options)
+      : _setupPrincipalListWidget(options);
```

Comment:
Is this return value used for anything? If not, let's just use a normal if/else.

Reply:
Yes, it isn't.